torch/utils/data/dataset.py

```diff
@@ -36,7 +36,14 @@ def __init__(self, data_tensor, target_tensor):
         self.target_tensor = target_tensor
 
     def __getitem__(self, index):
-        return self.data_tensor[index], self.target_tensor[index]
+        def select_and_keep_type(tensor, idx):
```

Comment:
nit: I think this is a bit inefficient because the function gets declared at every `__getitem__` call. Maybe it would be better to move it outside of `__getitem__`?